src/components/drag-and-drop-list/DragAndDropList.tsx

```diff
@@ -0,0 +1,137 @@
+import {useRef, useState} from "react";
+import "./drag-and-drop-list.css";
+import {DragAndDropTodoItem} from "../drag-and-drop-todo-item/DragAndDropTodoItem";
+import {GhostItem} from "../ghost-item/GhostItem";
+
+interface DragAndDropListProps {
+  title: string
+  listItems: Array<string>
+  listId: number
+  setListItems: (listItems: Array<string>) => void;
+  onItemDropped?: (item: string, sourceListId: number, targetListId: number) => void;
+}
+interface Coordinates {
+  x: number
+  y: number
+}
+
+export function DragAndDropList(props: DragAndDropListProps) {
+  const { title, listItems: items, setListItems: setItems, listId, onItemDropped } = props;
+
+  const [dragStartIdx, setDragStartIdx] = useState<number | null>(null);
+  const [isDragging, setIsDragging] = useState(false);
+  const [ghostPosition, setGhostPosition] = useState<Coordinates>({ x: 0, y: 0 });
+  const [draggedItem, setDraggedItem] = useState<string | null>(null);
+  const [hoverIndex, setHoverIndex] = useState(null);
+  const [isOverDraggableArea, setIsOverDraggableArea] = useState(false);
+
+  const droppableRef = useRef(null);
+
+  function handleDragStart(e, id: string) {
+    console.log(+id)
+    setDragStartIdx(+id)
+    setDraggedItem(items[+id]);
+    setIsDragging(true);
+    setGhostPosition({ x: e.clientX, y: e.clientY });
+    e.dataTransfer.setData('text/plain', JSON.stringify({id, name: items[+id], listId}));
+
+    const transparentImage = new Image();
+    transparentImage.src = "data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"; // 1x1 transparent GIF
+    e.dataTransfer.setDragImage(transparentImage, 0, 0);
+  };
+
+  function handleDrag(e) {
+    setGhostPosition({ x: e.clientX, y: e.clientY });
+  };
+
+  function handleDragEnd() {
+    setIsDragging(false);
+    setDraggedItem(null);
+  };
+
+  function handleDragOver(e, index) {
+    e.preventDefault();
+    setHoverIndex(index);
+    setIsOverDraggableArea(true);
+  };
+
+  function handleDrop(e) {
+    const json = e.dataTransfer.getData('text/plain')
+    const {id, name, listId: draggingItemsListId} = JSON.parse(json)
+
+    if(draggingItemsListId === listId) {
+      handleDropOnSameList(name, id)
+    } else {
+      handleDropOnDifferentList(name, +draggingItemsListId)
+    }
+  };
+
+  function handleDropOnDifferentList(name: string, sourceId: number) {
+    items.splice(hoverIndex, 0, name);
+    setItems([...items])
```

Comment:
This is a really bad practice. Basically, you are setting the state without setState with splice method. Than, you set it once more with setState. I would suggest this instead.
```suggestion
setItems(items.toSpliced(hoverIndex, 0, name))
```

Reply:
So splice method is build in JS method to modify items of an array in-place. Here i modify them in-place which does not cause react to rerender since react only rerenders when reference to the object of the setState() method changes not when the content of the array changes. For instance
```js
items.splice(hoverIndex, 0, name)
setItems(items)
```
will not cause rerender since reference to the items array did not change. So i clone the array and create a new one with spread operator so that react will rerender.
the suggested code does not work since there is no toSpliced() method in JS and if i were to try to write custom `toSplice` method i will be doing the same thing that i've already done so i dont see better way to handle this situation. Do you now any other way to handle without writing extra long code? 
Apart from that can you elaborate more, why this is bad practice? It seems to me perfectly okay way to handle this situation, and I am not able to see why this is a bad practice?